public/client-v2/src/components/CAPI/CAPITagQuery.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+import * as React from 'react';
+/* eslint-disable import/no-duplicates */
+import capiQuery from '../../services/capiQuery';
+import { type Fetch } from '../../services/capiQuery';
+/* eslint-enable import/no-duplicates */
+import Async, { type AsyncChild } from '../util/Async';
+
+type Tag = {
+  id: string,
+  webTitle: string,
+  webUrl: string
+};
+
+type CAPITagQueryReponse = {
+  response: {
+    results: Tag[]
+  }
+};
+
+type CAPITagQueryChild = AsyncChild<CAPITagQueryReponse>;
+
+type CAPISearchQueryProps = {
+  baseURL?: string,
+  fetch?: Fetch,
+  children: CAPITagQueryChild,
+  params: Object
+};
+
+class CAPIQuery extends React.Component<CAPISearchQueryProps> {
+  static defaultProps = {
+    params: {}
+  };
+
+  constructor(props: CAPISearchQueryProps) {
+    super(props);
+    this.setupCAPI(this.props.baseURL, this.props.fetch);
+  }
+
+  componentWillReceiveProps(nextProps: CAPISearchQueryProps) {
+    if (
+      (nextProps.baseURL && this.props.baseURL !== nextProps.baseURL) ||
+      (nextProps.fetch && this.props.fetch !== nextProps.fetch)
+    ) {
+      this.setupCAPI(this.props.baseURL, this.props.fetch);
+    }
+  }
+
+  setupCAPI(baseURL?: string, fetch?: Fetch): void {
+    this.capi = capiQuery(baseURL, fetch).tags;
+  }
+
+  capi: $ElementType<$Call<typeof capiQuery, string>, 'tags'>;
```

Comment:
I think we want to restrict the tag types we are searching here as they are not all relevant to fronts. The current tool also allows for choosing the section option to search for section tags but this can be added in another pr.